lib/web_ui/lib/src/engine/pointer_binding/event_position_helper.dart

```diff
@@ -19,14 +19,14 @@ import '../semantics.dart' show EngineSemanticsOwner;
 /// It also takes into account semantics being enabled to fix the case where
 /// offsetX, offsetY == 0 (TalkBack events).
 ui.Offset computeEventOffsetToTarget(DomMouseEvent event, DomElement actualTarget) {
-  // On top of a platform view
-  if (event.target != actualTarget) {
-    return _computeOffsetOnPlatformView(event, actualTarget);
-  }
   // On a TalkBack event
   if (EngineSemanticsOwner.instance.semanticsEnabled && event.offsetX == 0 && event.offsetY == 0) {
```

Comment:
@ditman is the conditional order ok to change here? Running into a TalkBack issue where the offsets aren't being calculated correctly for pointer events.  The problem is arising because any event that crosses into the shadowDOM has a `target` of whatever the shadowDOM's host is (glass pane).  With the semantic tree moved outside of the shadowDOM, the `target` is now the actual element being clicked, which was causing us to branch into this conditional, since `target` and `actualTarget` won't both be the glass pane. 

cc @yjbanov this change fixes the TalkBack issue you discovered in the Gallery app.

Reply:
I think this change makes sense. I'm not sure I have any special criteria when I added the ordering in the first place, but in this case it seems we're going from "most special corner case" to "least special case", so I think this change makes a lot of sense (also, glad that the change is simple and not something involving changing the measurements/algos :P)

Reply:
Spoke with @htoor3 over a VC. Some notes:

* It seems the `event.target == actualTarget` is never true because `actualTarget` is `<flt-glass-pane>`, but the shadow root host is the new `<flt-render-host>`.
* Simply changing to `<flt-render-host>` may not be sufficient because the semantics tree now behaves like a platform view (it's outside the shadow root) and may require similar logic to `_computeOffsetOnPlatformView`.

Reply:
Update on this:
* To your first point, this logic actually still does work because `<flt-render-host>` is invisible.  So when we hover/click on anything in the app, the `event.target` is still `<flt-glass-pane>` and the offset logic works correctly.  
* Since we're removing text and semantic nodes outside of the shadowDOM, those offsets now need to be calculated with the same logic as how platform view offsets were being calculated. This was already happening but it was a bit confusing as to why.  The `event.target != actualTarget` expression is actually a good check for this because it doesn't seem like we need to differentiate between nodes that are platform views and other non-shadowDOM nodes - we just need to determine if any given target is outside the shadowDOM.  If so, we need to recalculate the offsets. 
* To make this clearer, I renamed the boolean to check for this to `isTargetOutsideOfShadowDOM` and the function to a more generalized name `_computeOffsetRelativeToActualTarget` since it's computing offsets for more than just platform views now. 
* Tested this in various corner cases and the offsets are working correctly for:
    * Text nodes
    * Platform views
    * Semantic tree nodes
    * When the page is scrolled
    * When the app is embedded into a `<div>`
    * Using screen readers (both VoiceOver and Talkback)